### ISSUE TITLE
Failed to install.

### ISSUE THREAD

**Issue by NeilAllen** [dotnet/upgrade-assistant#1015]
Tool 'upgrade-assistant' failed to update due to the following:
Failed to install tool package 'upgrade-assistant': Access to the path 'C:\Users\<username>\.dotnet\tools\.store\.stage\1jvuevgm.cbo' is denied.
Tool 'upgrade-assistant' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool

**Comment by sunandabalu** [comment]
@NeilAllen Could you try installing in a different folder? looks like the one you are trying has some permission issues.

**Comment by NeilAllen** [comment]
> @NeilAllen Could you try installing in a different folder? looks like the one you are trying has some permission issues.

I'm using: dotnet tool install -g upgrade-assistant

How should I install to a different folder?

Thanks for your time.

**Comment by sunandabalu** [comment]
I was referring to the error message pasted above - `Failed to install tool package 'upgrade-assistant': Access to the path 'C:\Users<username>.dotnet\tools.store.stage\1jvuevgm.cbo' is denied.` Try using the option `--tool-path <path>` along with the `dotnet tool install` command to try installing in a different folder. https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-tool-install#description

**Comment by NeilAllen** [comment]
@sunandabalu Disappointingly, I get the same issue - even from an elevated command prompt:

# dotnet tool install upgrade-assistant --tool-path d:\nuget
Failed to install tool package 'upgrade-assistant': Access to the path 'd:\nuget\.store\.stage\5i0skqcq.q3v' is denied.
Tool 'upgrade-assistant' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool

**Comment by NeilAllen** [comment]
@sunandabalu 

Yes, if I follow the ms botsay tutorial, I can install the package OK.
 
dotnet tool install --add-source ./microsoft.botsay/nupkg microsoft.botsay    
You can invoke the tool from this directory using the following commands: 'dotnet tool run botsay' or 'dotnet botsay'.
Tool 'microsoft.botsay' (version '1.0.0') was successfully installed. Entry is added to the manifest file M:\microsoft.botsay\repository\.config\dotnet-tools.json.

**Comment by NeilAllen** [comment]
After looking at the working install, for botsay, I was able to figure out how to use a local source for the downloaded .nupkg file:

dotnet tool install --add-source D:/Downloads upgrade-assistant               
You can invoke the tool from this directory using the following commands: 'dotnet tool run upgrade-assistant' or 
'dotnet upgrade-assistant'.
Tool 'upgrade-assistant' (version '0.3.261602') was successfully installed. Entry is added to the manifest file

**Comment by NeilAllen** [comment]
See above

**Comment by NeilAllen** [comment]
Had to get back on this 1.

The real issue was McAfee Endpoint Security Scanners - I disabled it and `dotnet install tool update-assistant` works straight out of the box.

It seems mcafee is problematic elsewhere: 
   https://ryanhayes.net/how-to-run-docker-windows-containers-with-mcafee-endpoint-security/
   https://stackoverflow.com/questions/41104410/pull-image-with-docker-on-windows-10-failed-to-register-layer-denied-acce

It's not really a suitable product for development - mcafee - clearly.

**Comment by carlin-q-scott** [comment]
I experienced this issue and I was able to finally install by copying `%USERPROFILE%\.dotnet\tools\.store\.stage\randomfolder` to my download folder while the install was running. I didn't run the install manually from that folder. Literally the process of Windows copying the folder while the tool installation was running, caused it to succeed. 

I think Windows prevented the install script from deleting the staging folder. Or maybe Webroot Endpoint Protection stopped interfering with the installation. But I kind of doubt that because I tried installing 8 times before I succeeded. I also uninstalled and reinstalled without copying the folder, and it failed once again. Then I tried once more while copying the folder, and it succeeded yet again.

[upgrade assistant install attempts.log](https://github.com/dotnet/upgrade-assistant/files/11216816/upgrade.assistant.install.attempts.log)
